frontends/web/src/components/backups/restore.tsx

```diff
@@ -14,37 +14,53 @@
  * limitations under the License.
  */
 
-import { Component, h } from 'preact';
+import { Component, h, RenderableProps } from 'preact';
 import { route } from 'preact-router';
-import { translate } from 'react-i18next';
-import { Button, Checkbox } from '../forms';
-import { Dialog } from '../dialog/dialog';
-import WaitDialog from '../wait-dialog/wait-dialog';
-import Spinner from '../spinner/Spinner';
-import { PasswordRepeatInput } from '../password';
+import { translate, TranslateProps } from '../../decorators/translate';
 import { apiPost } from '../../utils/request';
 import { alertUser } from '../alert/Alert';
+import { Dialog } from '../dialog/dialog';
+import { Button, Checkbox } from '../forms';
+import { PasswordRepeatInput } from '../password';
+import Spinner from '../spinner/Spinner';
+import WaitDialog from '../wait-dialog/wait-dialog';
 import * as style from './backups.css';
 
-@translate()
-export default class Restore extends Component {
-    state = {
+interface RestoreProps {
+    selectedBackup: string;
+    requireConfirmation: boolean;
+    deviceID: string;
+    onRestore: () => void;
+}
+
+type Props = RestoreProps & TranslateProps;
+
+interface State {
+    isConfirming: boolean;
+    activeDialog: boolean;
+    isLoading: boolean;
+    understand: boolean;
+    password: string | null;
```

Comment:
Using `undefined` instead of `null` (and then declaring it just as `password?: string;`) would be the TypeScript way to do this in my opinion but I don't mind how it is now.

Reply:
TIL! Thx, fixed.

Reply:
`password?: string;` 👍